src/test/playwright/e2e/exercise/programming/ProgrammingExerciseParticipation.spec.ts

```diff
@@ -30,31 +37,132 @@ test.describe('Programming exercise participation', () => {
+                if (process.env.CI === 'true') {
+                    repoUrl = repoUrl.replace('localhost', 'artemis-app');
+                }
+                repoUrl = repoUrl.replace(studentOne.username!, `${studentOne.username!}:${studentOne.password!}`);
+                const urlParts = repoUrl.split('/');
+                const repoName = urlParts[urlParts.length - 1];
+                const exerciseRepo = await gitClient.cloneRepo(repoUrl, repoName);
+                const submission = javaPartiallySuccessfulSubmission;
+                const commitMessage = 'Initial implementation';
+                await makeGitSubmission(exerciseRepo, repoName, studentOne, submission, commitMessage);
+                await fs.rmdir(`./test-exercise-repos/${repoName}`, { recursive: true });
+                await page.goto(`courses/${course.id}/exercises/${exercise.id!}`);
+                const resultScore = await programmingExerciseOverview.getResultScore();
                 await expect(resultScore.getByText(submission.expectedResult)).toBeVisible();
             });
-        });
 
-        test('Makes a successful submission', async ({ programmingExerciseEditor }) => {
-            await programmingExerciseEditor.startParticipation(course.id!, exercise.id!, studentThree);
-            const submission = javaAllSuccessfulSubmission;
-            await programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, async () => {
-                const resultScore = await programmingExerciseEditor.getResultScore();
+            test('Makes a successful submission', async ({ page, programmingExerciseOverview }) => {
+                await programmingExerciseOverview.startParticipation(course.id!, exercise.id!, studentOne);
+                let repoUrl = await programmingExerciseOverview.getRepoUrl();
+                if (process.env.CI === 'true') {
+                    repoUrl = repoUrl.replace('localhost', 'artemis-app');
+                }
+                repoUrl = repoUrl.replace(studentOne.username!, `${studentOne.username!}:${studentOne.password!}`);
+                console.log('Repo URL in UI: ' + repoUrl);
+                const urlParts = repoUrl.split('/');
+                const repoName = urlParts[urlParts.length - 1];
+                const exerciseRepo = await gitClient.cloneRepo(repoUrl, repoName);
+                const submission = javaAllSuccessfulSubmission;
+                const commitMessage = 'Implemented all tasks';
+                await makeGitSubmission(exerciseRepo, repoName, studentOne, submission, commitMessage);
+                await fs.rmdir(`./test-exercise-repos/${repoName}`, { recursive: true });
+                await page.goto(`courses/${course.id}/exercises/${exercise.id!}`);
+                const resultScore = await programmingExerciseOverview.getResultScore();
                 await expect(resultScore.getByText(submission.expectedResult)).toBeVisible();
```

Comment:
This test differs very little from the one before it. Can you parameterize it & maybe extract common functionality to a method since other programming languages also follow this pattern?